2021-12-09T15:42:47|INFO|HydraHarpOptimizerMeasure|hydraharp_channel_optimizer start_stop False
2021-12-09T15:42:47|INFO|HydraHarpOptimizerMeasure|measurement hydraharp_channel_optimizer stopping stop_first
2021-12-09T15:42:47|INFO|HydraHarpHistogramMeasure|hydraharp_histogram start_stop False
2021-12-09T15:42:47|INFO|HydraHarpHistogramMeasure|measurement hydraharp_histogram stopping stop_first
2021-12-09T15:42:47|INFO|PowerScanMeasure|power_scan start_stop False
2021-12-09T15:42:47|INFO|PowerScanMeasure|measurement power_scan stopping stop_first
2021-12-09T15:42:47|INFO|PowerMeterOptimizerMeasure|powermeter_optimizer start_stop False
2021-12-09T15:42:47|INFO|PowerMeterOptimizerMeasure|measurement powermeter_optimizer stopping stop_first
2021-12-09T15:42:47|INFO|Sequencer|sequencer start_stop False
2021-12-09T15:42:47|INFO|Sequencer|measurement sequencer stopping stop_first
2021-12-09T15:42:47|INFO|APDOptimizerCBMeasurement|apd_optimizer start_stop False
2021-12-09T15:42:47|INFO|APDOptimizerCBMeasurement|measurement apd_optimizer stopping stop_first
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for hydraharp_channel_optimizer measurement hydraharp_channel_optimizer
2021-12-09T15:42:47|INFO|HydraHarpOptimizerMeasure|hydraharp_channel_optimizer start_stop False
2021-12-09T15:42:47|INFO|HydraHarpOptimizerMeasure|measurement hydraharp_channel_optimizer stopping stop_first
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for hydraharp_histogram measurement hydraharp_histogram
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for power_scan measurement power_scan
2021-12-09T15:42:47|INFO|PowerScanMeasure|power_scan start_stop False
2021-12-09T15:42:47|INFO|PowerScanMeasure|measurement power_scan stopping stop_first
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for powermeter_optimizer measurement powermeter_optimizer
2021-12-09T15:42:47|INFO|PowerMeterOptimizerMeasure|powermeter_optimizer start_stop False
2021-12-09T15:42:47|INFO|PowerMeterOptimizerMeasure|measurement powermeter_optimizer stopping stop_first
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for sequencer measurement sequencer
2021-12-09T15:42:47|INFO|Sequencer|sequencer start_stop False
2021-12-09T15:42:47|INFO|Sequencer|measurement sequencer stopping stop_first
2021-12-09T15:42:47|WARNING|HydraHarpOptimizerMeasure|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|WARNING|HydraHarpHistogramMeasure|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|WARNING|PowerScanMeasure|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|WARNING|PowerMeterOptimizerMeasure|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|WARNING|Sequencer|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|WARNING|APDOptimizerCBMeasurement|Measurement.gui is deprecated, use Measurement.app <class 'DeprecationWarning'>
2021-12-09T15:42:47|INFO|NearFieldMicroscope|setting up figures for apd_optimizer measurement apd_optimizer
2021-12-09T15:42:47|INFO|APDOptimizerCBMeasurement|apd_optimizer start_stop False
2021-12-09T15:42:47|INFO|APDOptimizerCBMeasurement|measurement apd_optimizer stopping stop_first
2021-12-09T15:42:48|INFO|NearFieldMicroscope|ini settings loading from near_field_microscope_defaults.ini
2021-12-09T15:42:48|DEBUG|FileLQ|save_dir: update_value 'C:\\Users\\RAMAN\\workspace\\NearField\\NearFieldMicroscope\\data' --> 'E:\\Natalie'    sender=None
2021-12-09T15:42:48|DEBUG|FileLQ|save_dir: different values C:\Users\RAMAN\workspace\NearField\NearFieldMicroscope\data E:\Natalie
2021-12-09T15:42:48|INFO|NearFieldMicroscope|hardware/hydraharp
2021-12-09T15:42:48|INFO|NearFieldMicroscope|hardware/pololu_maestro
2021-12-09T15:42:51|INFO|NearFieldMicroscope|hardware/power_wheel
2021-12-09T15:42:51|INFO|NearFieldMicroscope|hardware/thorlabs_powermeter
2021-12-09T15:42:51|INFO|NearFieldMicroscope|hardware/thorlabs_stepper_controller
2021-12-09T15:42:51|INFO|NearFieldMicroscope|hardware/apd_counter
2021-12-09T15:42:51|INFO|NearFieldMicroscope|ini settings loaded from: near_field_microscope_defaults.ini
2021-12-09T15:43:09|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 112, in __init__
    self._id = self._serial_numbers[dev_num]
IndexError: list index out of range

2021-12-09T15:43:54|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 122, in __init__
    _err(S.SBC_Open(self._id))
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 66, in _err
    raise IOError("Thorlabs Kinesis Error [{}] {}: {} ".format(retval, err_name, description))
OSError: Thorlabs Kinesis Error [3] FT_DeviceNotOpened: The Device must be opened before it can be accessed. See the appropriate Open function for your device. 

2021-12-09T15:44:30|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 122, in __init__
    _err(S.SBC_Open(self._id))
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 66, in _err
    raise IOError("Thorlabs Kinesis Error [{}] {}: {} ".format(retval, err_name, description))
OSError: Thorlabs Kinesis Error [3] FT_DeviceNotOpened: The Device must be opened before it can be accessed. See the appropriate Open function for your device. 

2021-12-09T15:44:44|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|ThorlabsStepperControllerDev.__init__
2021-12-09T15:44:45|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_numbers available b'' --> []
2021-12-09T15:44:45|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_number requested '55902790'
2021-12-09T15:44:45|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|using serial number b'55902790'
2021-12-09T15:44:45|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|SBC_Open: b'55902790'
2021-12-09T15:44:45|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 122, in __init__
    _err(S.SBC_Open(self._id))
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 66, in _err
    raise IOError("Thorlabs Kinesis Error [{}] {}: {} ".format(retval, err_name, description))
OSError: Thorlabs Kinesis Error [3] FT_DeviceNotOpened: The Device must be opened before it can be accessed. See the appropriate Open function for your device. 

2021-12-09T15:44:47|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|ThorlabsStepperControllerDev.__init__
2021-12-09T15:44:47|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_numbers available b'' --> []
2021-12-09T15:44:47|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_number requested '55902790'
2021-12-09T15:44:47|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|using serial number b'55902790'
2021-12-09T15:44:47|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|SBC_Open: b'55902790'
2021-12-09T15:44:47|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 122, in __init__
    _err(S.SBC_Open(self._id))
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 66, in _err
    raise IOError("Thorlabs Kinesis Error [{}] {}: {} ".format(retval, err_name, description))
OSError: Thorlabs Kinesis Error [3] FT_DeviceNotOpened: The Device must be opened before it can be accessed. See the appropriate Open function for your device. 

2021-12-09T15:47:12|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|ThorlabsStepperControllerDev.__init__
2021-12-09T15:47:12|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_numbers available b'' --> []
2021-12-09T15:47:12|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|serial_number requested '55902790'
2021-12-09T15:47:12|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|using serial number b'55902790'
2021-12-09T15:47:12|DEBUG|ScopeFoundryHW.thorlabs_stepper_motors.thorlabs_stepper_controller_dev|SBC_Open: b'55902790'
2021-12-09T15:47:12|CRITICAL|root|Unhandled exception:Traceback (most recent call last):
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 220, in enable_connection
    raise err
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundry\hardware.py", line 210, in enable_connection
    self.connect()
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_hw.py", line 81, in connect
    debug=S['debug_mode'])
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 123, in __init__
    _err(S.SBC_Open(self._id))
  File "C:\Users\RAMAN\workspace\NearField\ScopeFoundryHW\thorlabs_stepper_motors\thorlabs_stepper_controller_dev.py", line 66, in _err
    raise IOError("Thorlabs Kinesis Error [{}] {}: {} ".format(retval, err_name, description))
OSError: Thorlabs Kinesis Error [3] FT_DeviceNotOpened: The Device must be opened before it can be accessed. See the appropriate Open function for your device. 

